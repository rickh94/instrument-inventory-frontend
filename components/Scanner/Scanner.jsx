import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import Quagga from 'quagga'

import './Scanner.scss'

function Scanner(props) {
  useEffect(() => {
    function _onDetected(result) {
      props.onDetected(result)
    }
    Quagga.init(
      {
        inputStream: {
          type: 'LiveStream',
          contstraints: {
            width: { min: 640 },
            height: { min: 480 },
            aspectRatio: { min: 1, max: 100 },
            facing: 'environment',
          },
        },
        locator: {
          patchSize: 'medium',
          halfSample: true,
        },
        numOfWorkers: 2,
        decoder: {
          readers: ['code_128_reader', 'code_39_reader'],
        },
        locate: true,
      },
      err => {
        if (err) {
          return console.error(err)
        }
        Quagga.start()
      }
    )
    Quagga.onDetected(_onDetected)
    return function cleanup() {
      Quagga.offDetected(_onDetected)
      Quagga.stop()
    }
  })

  return <div id="interactive" className="viewport" />
}

Scanner.propTypes = {
  onDetected: PropTypes.func.isRequired,
}

export default Scanner
