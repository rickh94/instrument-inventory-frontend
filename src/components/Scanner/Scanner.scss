@charset "UTF-8";

#interactive.viewport {
    position: relative;
}

#interactive.viewport > canvas, #interactive.viewport > video {
    max-width: 100%;
    width: 100%;
}

canvas.drawing, canvas.drawingBuffer {
    position: absolute;
    left: 0;
    top: 0;
}

@media (max-width: 603px) {
  /* line 5, ../sass/phone/_viewport.scss */
  #interactive.viewport {
    width: 100%;
    height: auto;
    overflow: hidden;
  }

}
