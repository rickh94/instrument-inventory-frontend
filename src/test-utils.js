/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme, indigo } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

const theme = createMuiTheme({
  palette: {
    primary: indigo,
    secondary: {
      main: '#0A81D1',
    },
  },
})

export const showAlert = jest.fn()
export const setSearchResults = jest.fn()
export const setFilterResults = jest.fn()
export const searchResults = jest.fn()
export const filterResults = jest.fn()

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
)

export const TestHelpers = ({ children }) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children }) => (
  <TestRouter>
    <TestTheme>
      <TestHelpers>{children}</TestHelpers>
    </TestTheme>
  </TestRouter>
)

// eslint-disable-next-line no-undef
export const flushPromises = () => new Promise(setImmediate)

const customRender = (ui, options, schema = { components: { schemas: {} } }) =>
  render(<TestSchema schema={schema}>{ui}</TestSchema>, {
    wrapper: TestEverything,
    ...options,
  })

export * from '@testing-library/react'

export { customRender as render }
