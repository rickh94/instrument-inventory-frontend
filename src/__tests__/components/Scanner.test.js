import React from 'react'
import Scanner from '../../components/Scanner/Scanner'
import { render } from '../../test-utils'
import Quagga from 'quagga'

describe('<Scanner />', () => {
  it('matches snapshot', () => {
    Quagga.init = jest.fn()
    Quagga.onDetected = jest.fn()
    Quagga.stop = jest.fn()
    const onDetectedMock = jest.fn()
    const { container } = render(<Scanner onDetected={onDetectedMock} />)
    expect(container).toMatchSnapshot()
    expect(Quagga.init).toHaveBeenCalled()
    expect(Quagga.onDetected).toHaveBeenCalledWith(onDetectedMock)
  })
})
