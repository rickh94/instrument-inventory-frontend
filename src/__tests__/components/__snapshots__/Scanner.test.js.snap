// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scanner /> matches snapshot 1`] = `
<div>
  <div
    class="viewport"
    data-testid="scanner"
    id="interactive"
  />
</div>
`;
